### PR TITLE
dev: generate assets/github-action-config.json

### PR DESCRIPTION
**Pull request by jirfag**
It will be used by GitHub action `golangci-lint-action`.

Relates: golangci/golangci-lint-action#11

### CODE REVIEW

File: scripts/gen_github_action_config/main.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	if err = enc.Encode(cfg); err != nil {
+		return fmt.Errorf("failed to json encode config: %w", err)
+	}
+
+	return nil
+}
+
+type logInfo struct {
+	Warning string `json:",omitempty"`
+	Info    string `json:"omitempty"`
```
Comment (ernado) [inline scripts/gen_github_action_config/main.go:53]: Seems like a typo, probably `,omitempty` is expected here

File: scripts/gen_github_action_config/main.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	if len(os.Args) != 2 { //nolint:gomnd
+		return fmt.Errorf("usage: go run .../main.go out-path.json")
+	}
+	outFile, err := os.Create(os.Args[1])
+	if err != nil {
+		return fmt.Errorf("failed to create output config file: %w", err)
+	}
+	defer outFile.Close()
```
Comment (ernado) [inline scripts/gen_github_action_config/main.go:41]: Probably we still want to check this error.

Comment (jirfag) [reply scripts/gen_github_action_config/main.go:41]: it's just a script, we can visually validate if it's output is corrupted.

File: scripts/gen_github_action_config/main.go (hunk 1 above)
Comment (jirfag) [reply scripts/gen_github_action_config/main.go]: you are right, thank you, we need a linter for that :)

**Pull request merged**